src/app/app-routing.module.ts

```diff
@@ -4,16 +4,25 @@ import { RouterModule, Routes } from '@angular/router';
 import { CoursesPageComponent } from '@app/courses-page';
 import { LoginPageComponent } from '@app/login-page';
 import { NewCoursesPageComponent } from '@app/new-courses-page';
+import { ErrorPageComponent } from '@app/error-page';
+import { AuthGuard } from '@app/core';
 
 const routes: Routes = [
-  { path: 'courses', component: CoursesPageComponent },
+  { 
+    path: 'courses',
+    canActivate: [AuthGuard],
+    children: [
+    { path: '', component: CoursesPageComponent },
```

Comment:
move those to a separate module and load them lazily via loadChildren

https://angular.io/guide/lazy-loading-ngmodules

Reply:
Changed to lazy loading